### DISCUSSION TITLE
Redirecting old wiki pages

### DISCUSSION THREAD

**Discussion by mhauru** [alan-turing-institute/REG-handbook#42] [General]
Hello handbook! I'm going over the contents of the handbook right now for the first time, and it's looking really good. Thank you so much for all the work over the hackweek.

One request I would have is that when a handbook page has been created that replaces a wiki page, rather than removing the old wiki page, we would replace it with a page that links to the handbook. This came up with the onboarding page, because some links from within the wiki and elsewhere were still pointing to it, but it no longer existed. I changed it to look like this: https://github.com/alan-turing-institute/research-engineering-group/wiki/New-Joiners:-Start-here

I'll go over the onboarding section and update it with the latest practices. Once that's done, I'll try to fix all links to point to the handbook rather than the wiki, but I would still leave the above page alive as a safety measure.

**Comment by JimMadge** [comment]
I had thought about this, I can't quite decide what is worse.

Points against each approach,

Removing pages immediately

- Dead ends might prevent someone from finding what they are looking for
- Risk of loosing information?

Keeping migrated pages

- The wiki is left with a lot of stub pages
- We have to decide when to remove these pages (when have all links been updated?)
- A working link to a 'redirection' page might not be a strong prompt to update links

**Reply by mhauru** [reply]
I see the value in all of your bullet points, but for me the "Dead ends might prevent someone from finding what they are looking for" part outweighs the others. If we manage to remove all links to a wiki page, then the list of all wiki pages is the only place where it would appear and increase clutter. And if we don't manage to remove all links and somebody stumbles upon the stub page, then I think it's always better to point them to where they should be, rather than frustrate them with a dead link. They don't even know why it's dead and may not know where to look instead. A link that says "New joiners: Start here" but leads to a dead end isn't a great experience on a first day.

**Reply by JimMadge** [reply]
:+1:

I think the important thing is then to just make it clear that the stub or legacy pages shouldn't be used to add new information, to avoid duplication or splitting up topics.

Maybe later we can have a spring clean to remove unneeded wiki pages when it is more reasonable.

**Comment by andrewphilipsmith** [comment]
I think a stub page is a better, albeit imperfect, user experience than a dead link.

To encourage others to update external links I think that a redirect page with a time delay is the least-worst option. ("This page will redirect to http://example.com in 5 seconds" kinda thing). That said I've never tried to create one using GH-flavoured markdown.
